services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedRegistrationService.java

```diff
@@ -196,32 +198,32 @@ public void setConfig(final MongoDbBasedRegistrationConfigProperties config) {
         return processResolveGroupMembers(tenantId, viaGroups, span);
     }
 
-    private JsonObject convertDevice(final String deviceId, final Device payload) {
-
-        if (payload == null) {
-            return null;
-        }
-
-        final JsonObject data = JsonObject.mapFrom(payload);
-
-        return new JsonObject()
-                .put(RegistryManagementConstants.FIELD_PAYLOAD_DEVICE_ID, deviceId)
-                .put("data", data);
+    private Future<DeviceDto> findDevice(final String tenantId, final String deviceId) {
+        return findDeviceDocument(tenantId, deviceId)
+                .compose(result -> Optional.ofNullable(result)
+                        .map(ok -> result.mapTo(DeviceDto.class))
+                        .map(Future::succeededFuture)
+                        .orElseGet(() -> Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_NOT_FOUND,
+                                String.format("Device [%s] not found.", deviceId)))));
     }
 
-    private Future<DeviceDto> findDevice(final String tenantId, final String deviceId) {
+    private Future<JsonObject> findDeviceDocument(final String tenantId, final String deviceId) {
         final JsonObject findDeviceQuery = new MongoDbDocumentBuilder()
                 .withTenantId(tenantId)
                 .withDeviceId(deviceId)
                 .document();
         final Promise<JsonObject> readDevicePromise = Promise.promise();
         mongoClient.findOne(config.getCollectionName(), findDeviceQuery, null, readDevicePromise);
-        return readDevicePromise.future()
-                .compose(result -> Optional.ofNullable(result)
-                        .map(ok -> result.mapTo(DeviceDto.class))
-                        .map(Future::succeededFuture)
-                        .orElseGet(() -> Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_NOT_FOUND,
-                                String.format("Device [%s] not found.", deviceId)))));
+        return readDevicePromise.future();
+    }
+
+    private RegistrationResult getRegistrationResult(final String deviceId, final JsonObject devicePayload) {
+        return RegistrationResult.from(HttpURLConnection.HTTP_OK,
+                Optional.ofNullable(devicePayload)
+                        .map(ok -> new JsonObject()
+                                .put(RegistryManagementConstants.FIELD_PAYLOAD_DEVICE_ID, deviceId)
```

Comment:
IMHO this should be `RegistrationConstants.FIELD_PAYLOAD_DEVICE_ID` as the result being prepared here is defined by the Device Registration API